lib/src/components/NxTransferList/types.ts

```diff
@@ -14,53 +14,69 @@ export type FilterFn<T extends string | number = string> = (d: DataItem<T>[]) =>
+  showReorderingButtons?: boolean | null;
+  onReorderItem?: ReorderSelectectedItemFunction | null;
+  index: number;
+  listLength: number;
 }
 
 export interface TransferListHalfProps<T extends string | number = string> {
   label: Exclude<ReactNode, null | undefined>;
   filterValue: string;
   onFilterChange: NxFilterInputProps['onChange'];
   showMoveAll: boolean;
-  onMoveAll: (toMove: Set<T>) => void;
+  onMoveAll: (toMove: T[]) => void;
   items: DataItem<T>[];
   isSelected: boolean;
   onItemChange: SelectionChangeHandler<T>;
   footerContent: ReactNode;
   filterFn?: ((filterStr: string, itemDisplayName: string) => boolean) | null;
+  onReorderItem?: ReorderSelectectedItemFunction | null;
+  allowReordering?: boolean | null;
 }
 
-export interface StatefulProps<T extends string | number = string>
+export interface BaseStatefulProps<T extends string | number = string>
   extends Omit<HTMLAttributes<HTMLDivElement>, 'children' | 'onChange'> {
   allItems: DataItem<T>[];
-  selectedItems: Set<T>;
   availableItemsLabel?: ReactNode;
   selectedItemsLabel?: ReactNode;
-  availableItemsCountFormatter?: ((n: number) => string) | null,
-  selectedItemsCountFormatter?: ((n: number) => string) | null,
+  availableItemsCountFormatter?: ((n: number) => string) | null;
+  selectedItemsCountFormatter?: ((n: number) => string) | null;
   showMoveAll?: boolean | null;
-  onChange: (newSelected: Set<T>) => void;
   filterFn?: ((filterStr: string, itemDisplayName: string) => boolean) | null;
 }
 
-export interface Props<T extends string | number = string> extends StatefulProps<T> {
+interface UnorderedStatefulProps<T extends string | number = string> extends BaseStatefulProps<T> {
+  allowReordering?: false | null;
+  selectedItems: Set<T>;
+  onChange: (newSelected: Set<T>) => void;
+}
+
+interface OrderedStatefulProps<T extends string | number = string> extends BaseStatefulProps<T> {
+  allowReordering: true;
+  selectedItems: T[];
+  onChange: (newSelected: T[]) => void;
+}
+
+export type StatefulProps<T extends string | number = string> =
+    UnorderedStatefulProps<T> | OrderedStatefulProps<T>;
+
+export type Props<T extends string | number = string> = StatefulProps<T> & {
   availableItemsFilter: string;
   selectedItemsFilter: string;
   onAvailableItemsFilterChange: NxFilterInputProps['onChange'];
   onSelectedItemsFilterChange: NxFilterInputProps['onChange'];
-}
+};
 
-export const propTypes: ValidationMap<Props<string | number>> = {
-  allItems: PropTypes.arrayOf(PropTypes.shape({
-    id: PropTypes.oneOfType([
-      PropTypes.string.isRequired,
-      PropTypes.number.isRequired
-    ]).isRequired,
-    displayName: PropTypes.string.isRequired
-  }).isRequired).isRequired,
-  selectedItems: PropTypes.instanceOf<Set<string | number>>(Set).isRequired,
+// allItems, selectedItems, onAvailableItemsFilterChange, onSelectedItemsFilterChange
+// are excluded in the propTypes due to clash with parametric Props;
```

Comment:
We should check whether this is still the case now that the types have been redone

Reply:
> We should check whether this is still the case now that the types have been redone

I was able to bring 2 back without errors in this commit:
ced6b8e23042c0886a1255efc67fb2f75f92812f